ocaml/tests/test_repository_helpers.ml

```diff
@@ -1990,6 +2620,156 @@ module GuidanceSetResortGuidancesTest = Generic.MakeStateless (struct
       ]
 end)
 
+module ParseLineOfListUpdates = Generic.MakeStateless (struct
+  module Io = struct
+    type input_t = string list
+
+    type output_t = (Pkg.t * string) list
+
+    let string_of_input_t = Fmt.(str "%a" Dump.(list string))
+
+    let string_of_output_t =
+      Fmt.(str "%a" Dump.(list (pair (record @@ fields_of_pkg) string)))
+  end
+
+  let transform lines =
+    let _, updates =
+      List.fold_left parse_line_of_list_updates ((None, None, None), []) lines
+    in
+    updates
+
+  let tests =
+    `QuickAndAutoDocumented
+      [
+        ( [
+            "xenserver-status-report.noarch"
+          ; "         1.3.1-1.xs8         \
```

Comment:
It'd be good to understand why `yum` is outputting such inconsistent formats... but at least our parsing is looking bulletproof :+1:

Reply:
Indeed. I can't get the output manually. And I can't find a configuration of YUM to control that. But we did observe this from XAPI log.